apps/nextjs/src/env/client.mjs

```diff
@@ -14,10 +14,10 @@ export const formatErrors = (
     })
     .filter(Boolean);
 
-if (!_clientEnv.success) {
+if (_clientEnv.success === false) {
```

Comment:
this has been an issue for some configurations, not really sure why putting explicit `=== false` helps here but the success-state is now correctly inferred

Reply:
Huh, interesting. 

I ran across some unexpected booleans when `new Boolean()` and `Boolean()` were combined:
```ts
const myFalse = new Boolean(false);   // initial value of false
const g = Boolean(myFalse);           // initial value of true
```
[see MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#:~:text=If%20you%20specify%20any%20object%2C%20including%20a%20Boolean%20object%20whose%20value%20is%20false%2C%20as%20the%20initial%20value%20of%20a%20Boolean%20object%2C%20the%20new%20Boolean%20object%20has%20a%20value%20of%20true.)


But no idea if that's the case!